harness/determined/estimator/_estimator_trial.py

```diff
@@ -404,10 +409,10 @@ def __init__(
     def pre_execute_hook(
         cls: Type["EstimatorTrialController"],
         env: det.EnvContext,
-        hvd_config: horovod.HorovodContext,
+        use_horovod: bool,
```

Comment:
I can't recall anymore what your last version had here.  It seems like probably you want to use the new _DistributedBackend here maybe?  Seems to carry the information you're trying to portray here?

I'd consider this a "non-blocking" review comment.  Ignore it if you wish.